src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.cpp

```diff
@@ -44,61 +44,74 @@ void MagnetizedTovStar::pup(PUP::er& p) {
   p | vector_potential_amplitude_;
 }
 
-template <typename DataType>
-tuples::TaggedTuple<hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>>
-MagnetizedTovStar::variables(
-    const tnsr::I<DataType, 3>& coords,
-    tmpl::list<
-        hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>> /*meta*/,
-    const RadialVariables<DataType>& radial_vars) const {
-  const size_t num_pts = get_size(get<0>(coords));
-  auto magnetic_field =
-      make_with_value<tnsr::I<DataType, 3, Frame::Inertial>>(num_pts, 0.0);
-  using std::max;
-  const auto sqrt_det_spatial_metric =
-      get<gr::Tags::SqrtDetSpatialMetric<DataType>>(variables(
-          coords, tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataType>>{},
-          radial_vars));
-  for (size_t i = 0; i < num_pts; ++i) {
-    const double pressure = get_element(get(radial_vars.pressure), i);
-    if (LIKELY(get_element(radial_vars.radial_coordinate, i) > 1.0e-16)) {
-      if (pressure < cutoff_pressure_) {
+namespace magnetized_tov_detail {
+template <typename DataType, StarRegion Region>
+void MagnetizedTovVariables<DataType, Region>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3>*> magnetic_field,
+    const gsl::not_null<Cache*> cache,
+    hydro::Tags::MagneticField<DataType, 3> /*meta*/) const {
+  if constexpr (Region == StarRegion::Exterior) {
```

Comment:
the previous implementation was _much_ easier to understand. Can you please revert to that?

Reply:
ping

Reply:
I wasn't sure what you were suggesting here. If you are referring to the interior/exterior/center switch, then I thought this is actually quite nice, since in the exterior the magnetic field is zero, at the center it is the r->0 limit and in the interior it checks for the pressure cutoff, which is also how the documentation describes this. What do you think?

Reply:
I think the old one is easier to read. we went from 2 `if`s to 4/5. That's a lot of added logic.

Reply:
Ok I'll take another look tomorrow 👍

Reply:
ok reverted this